include/utf/utf.h

```diff
@@ -26,15 +26,20 @@ namespace utf
  * @param dest The string to append the codepoint to
  * @param codepoint The UTF-32 codepoint to append
  */
-inline void utf8_append_codepoint(std::string& dest, uint32_t codepoint)
+inline void utf8_append_codepoint(std::string& dest, UChar32 codepoint)
```

Comment:
Would users ever have to call this function ? (Wondering about converting to `UChar32`)